cfme/tests/services/test_service_rbac.py

```diff
@@ -0,0 +1,129 @@
+                                    for k in ['Access Rules for all Virtual Machines']])
+    group = new_group(appliance=appliance, role=role.name)
+    user = new_user(appliance=appliance, group=group, credential=new_credential)
+    yield (role, user)
+
+
+@pytest.yield_fixture(scope='function')
+def _myservice(appliance, catalog_item):
+    vm_name = catalog_item.provisioning_data["catalog"]["vm_name"]
+    catalog_item.create()
+    service_catalogs = ServiceCatalogs(appliance, catalog_item.catalog, catalog_item.name)
+    service_catalogs.order()
+    yield catalog_item.name, vm_name
+
+
+def test_service_rbac_no_permission(appliance, role_user_group):
+    """ Test service rbac without user permission """
+    role, user = role_user_group
+    error_message = ("The user's role is not authorized for any access, "
+                     "please contact the administrator!")
+    with error.expected(error_message):
+        appliance.server.login(user)
+
+
+CATALOG_MODULES = ['Catalogs', 'Orchestration Templates', 'Catalog Items', 'Service Catalogs']
+
+
+@pytest.mark.parametrize('permission', CATALOG_MODULES)
+def test_service_rbac_catalog_explorer(appliance, role_user_group, catalog,
+                                       catalog_item, permission):
+    """ Test service rbac with only catalog explorer module permissions"""
+    role, user = role_user_group
+    if permission == CATALOG_MODULES[3]:
+        role.update({
+            'product_features': [(['Everything'], True)] + [(['Everything'], False)] +
+                                [(['Everything', 'Services', 'Requests', ], True)] +
+                                [(['Everything', 'Services', 'Catalogs Explorer', k], True)
+                                 for k in ['Catalog Items', 'Service Catalogs', 'Catalogs']] +
+                                [(['Everything', 'Automation', 'Automate', 'Customization'], True)]
+        })
+    else:
+        role.update({
+            'product_features': [(['Everything'], True)] + [(['Everything'], False)] +
+                                [(['Everything', 'Services', 'Catalogs Explorer', k], True)
+                                 for k in [permission]]
+        })
+
+    with user:
+        if permission == CATALOG_MODULES[0]:
+            assert catalog.exists
+        elif permission == CATALOG_MODULES[1]:
+            collection = appliance.collections.orchestration_templates
+            template = collection.create(template_type="CloudFormation Templates",
+                                         template_name=fauxfactory.gen_alphanumeric(),
+                                         temp_type='Amazon CloudFormation',
+                                         description="Template description",
+                                         content=fauxfactory.gen_numeric_string())
+            assert template.exists
+            template.delete()
+        elif permission == CATALOG_MODULES[2]:
+            catalog_item.create()
```

Comment:
Please check here , catalog_item.create may not be necessary as it gets created in the fixture .
So just assert  if catalog.exists